File: packages/jsii-pacmak/test/expected.jsii-calc/java/src/main/java/software/amazon/jsii/tests/calculator/Isomorphism.java
package software.amazon.jsii.tests.calculator;

/**
 * Checks the "same instance" isomorphism is preserved within the constructor.
 * <p>
 * Create a subclass of this, and assert that <code>this.myself()</code> actually returns
 * <code>this</code> from within the constructor.
 * <p>
 * EXPERIMENTAL
 */
@javax.annotation.Generated(value = "jsii-pacmak")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.Isomorphism")
public abstract class Isomorphism extends software.amazon.jsii.JsiiObject {

    protected Isomorphism(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected Isomorphism(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    protected Isomorphism() {
        super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
        software.amazon.jsii.JsiiEngine.getInstance().createNewObject(this);
    }

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public @org.jetbrains.annotations.NotNull software.amazon.jsii.tests.calculator.Isomorphism myself() {
        return this.jsiiCall("myself", software.amazon.jsii.tests.calculator.Isomorphism.class);
    }

    /**
     * A proxy class which represents a concrete javascript instance of this type.
     */
    final static class Jsii$Proxy extends software.amazon.jsii.tests.calculator.Isomorphism {
        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
            super(objRef);
        }
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/kotlin/src/main/kotlin/software/amazon/jsii/tests/kotlin/calculator/Isomorphism.kt
package software.amazon.jsii.tests.kotlin.calculator

/**
 * Checks the "same instance" isomorphism is preserved within the constructor.
 * 
 * Create a subclass of this, and assert that `this.myself()` actually returns
 * `this` from within the constructor.
 */
@javax.annotation.Generated("jsii-pacmak")
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.kotlin.calculator.`$Module`::class, fqn = "jsii-calc.Isomorphism")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
abstract class Isomorphism : software.amazon.jsii.JsiiObject {
    constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(objRef)
    constructor(mode: software.amazon.jsii.JsiiObject.InitializationMode) : super(mode)

    constructor() : super(software.amazon.jsii.JsiiObject.InitializationMode.JSII) {
        software.amazon.jsii.JsiiEngine.getInstance().createNewObject(this)
    }

    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    open fun myself(): software.amazon.jsii.tests.kotlin.calculator.Isomorphism {
        return this.jsiiCall("myself", software.amazon.jsii.tests.kotlin.calculator.Isomorphism::class.java) ?: error("Method 'myself()' returned null value")
    }

    class `Jsii$Proxy` : software.amazon.jsii.tests.kotlin.calculator.Isomorphism {
        protected constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(objRef) {
        }

        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        override fun myself(): software.amazon.jsii.tests.kotlin.calculator.Isomorphism {
            return this.jsiiCall("myself", software.amazon.jsii.tests.kotlin.calculator.Isomorphism::class.java) ?: error("Method 'myself()' returned null value")
        }
    }
}
